packages/libs/deer/desert/src/token.rs

```diff
@@ -3,69 +3,202 @@ use core::fmt::{Debug, Display, Formatter};
 use deer::Number;
 
 // TODO: test
-#[derive(Debug, Copy, Clone, PartialEq)]
+// TODO: this should be `Copy`, but `Number` has no &'static constructor
+#[derive(Debug, Clone, PartialEq)]
 pub enum Token {
-    /// A serialized `bool`
+    /// A serialized [`bool`]
+    ///
+    /// ```
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&true, &[Token::Bool(true)])
+    /// ```
+    Bool(bool),
+
+    /// A serialized [`Number`]
+    ///
+    /// ```
+    /// use deer::Number;
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&Number::from(1), &[Token::Number(Number::from(1))])
+    /// ```
+    Number(Number),
+
+    /// A serialized [`u128`]
+    ///
+    /// this is a separate variant, as [`Number`] is unable to fully represent 128 bit values, this
+    /// means that u128/i128 need special support.
+    ///
+    /// ```
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&1u128, &[Token::U128(1)])
+    /// ```
+    U128(u128),
+
+    /// A serialized [`i128`]
+    ///
+    /// this is a separate variant, as [`Number`] is unable to fully represent 128 bit values, this
+    /// means that u128/i128 need special support.
+    ///
+    /// ```
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&1i128, &[Token::I128(1)])
+    /// ```
+    I128(i128),
+
+    /// A serialized [`usize`]
+    ///
+    /// this is a separate variant, because there is no guarantee about the width of a usize/isize,
+    /// it depends on the size to reference a memory address on the host system. There might be
+    /// systems that use 128 bit in the future that rusts supports, which means we too need to
+    /// support those, by special casing usize.
+    ///
+    /// ```
+    /// use deer_desert::{assert_tokens, Token};
+    ///
+    /// assert_tokens(&1usize, &[Token::USize(1)])
+    /// ```
+    USize(usize),
+
+    /// A serialized [`isize`]
+    ///
+    /// this is a separate variant, because there is no guarantee about the width of a usize/isize,
+    /// it depends on the size to reference a memory address on the host system. There might be
+    /// systems that use 128 bit in the future that rusts supports, which means we too need to
+    /// support those, by special casing usize.
```

Comment:
```suggestion
    /// support those, by special casing isize.
```

or alternatlive use the `usize/isize` way to write this as you've done for u128/i128

Reply:
fixed in [4835afc](https://github.com/hashintel/hash/pull/1516/commits/4835afc8f0f9666cccc226ce700714ea14596206)